src/ripple/basics/impl/PerfLogImp.cpp

```diff
@@ -0,0 +1,508 @@
+//------------------------------------------------------------------------------
+/*
+    This file is part of rippled: https://github.com/ripple/rippled
+    Copyright (c) 2018 Ripple Labs Inc.
+
+    Permission to use, copy, modify, and/or distribute this software for any
+    purpose  with  or without fee is hereby granted, provided that the above
+    copyright notice and this permission notice appear in all copies.
+
+    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
+    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
+    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+*/
+//==============================================================================
+
+#include <ripple/basics/BasicConfig.h>
+#include <ripple/basics/impl/PerfLogImp.h>
+#include <ripple/beast/core/CurrentThreadName.h>
+#include <ripple/beast/utility/Journal.h>
+#include <ripple/json/json_writer.h>
+#include <ripple/json/to_string.h>
+#include <ripple/rpc/impl/Handler.h>
+#include <atomic>
+#include <cstdint>
+#include <cstdlib>
+#include <iostream>
+#include <stdexcept>
+#include <string>
+#include <unordered_map>
+#include <utility>
+
+using namespace std::chrono;
+using steady_time_point = time_point<steady_clock>;
```

Comment:
There are folks who say that global-scope `using` declarations are okay in a `*.cpp` file is okay.  That's because no one ever does a `#include` of a `*.cpp` file, so the infection stops here.

However our world is not that simple, because we have this unity build thing.  For unity builds there are `#include`s of `*.cpp` files.

So I suggest that these `using` declarations be should be removed from global scope as well.

That's a little bit more of a pain, because you'll either need to re-inject them into the code scopes (in this file) that need them, or you can simply fully spell out the fully qualified name, like you were doing before.

Thanks.